xformers/benchmarks/benchmark_mem_eff_attention.py

```diff
@@ -5,172 +5,236 @@
 
 
 import itertools
-import pprint
-from typing import Dict
+import math
+from functools import partial
+from typing import cast
 
 import torch
 from torch.utils import benchmark
+from utils import benchmark_main_helper
 
 import xformers.ops
 
+torch.backends.cuda.matmul.allow_tf32 = False
```

Comment:
not directly related, but did you get to test out the perf effect of tf32 accumulation on A100 ? asking just in case to learn a bit more

Reply:
The CUTLASS-based kernels do use a trick that benefit from `tf32` while having the fp32 numerics (by performing 3 matmuls in tf32 to leverage tensorcores). The trick (which was present in cutlass examples) was to decompose a fp32 tensor in `fp32 = (fp32_low_bits + fp32_high_bits)`, where `fp32_low_bits` and `fp32_high_bits` are tf32, so that the multiplication can be approximated by 3 matmuls on tf32 (and dropping the `low_bits * low_bits part`).

The implementation that uses only a single tf32 instruction is not implemented yet, but we were thinking it could be exposed by reading the info from `torch.backends.cuda.matmul.allow_tf32` and dispatching to different kernels.